### PR TITLE
Add book tests

### PR DESCRIPTION
**Pull request by storresy**
**Summary**
- Rspec, Factory bot and Faker gems added
- Database cleaner gem added and configured
- Book factory added
- Book factory test added
- Book validation tests added
- Configured travis to run specs

**Trello card**
https://trello.com/c/Fm4yvVQE

**Screenshots**
screen shot 2018-12-10 at 11 56 13

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ before_script:
 script:
   # - bundle exec rspec spec -fd
   - bundle exec rubocop app -R --format simple
+  - bundle exec rspec
```
Comment (Gmarn) [inline .travis.yml]: You can add the option `-fd` allow easier debug ;)

File: Gemfile [hunk 2]
```diff
@@ -45,6 +45,9 @@ group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
   gem 'rubocop', '~> 0.61.1'
+  gem 'rspec-rails', '~> 3.8'
```
Comment (Gmarn) [inline Gemfile:48]: This gem should be in you `test` group

File: Gemfile [hunk 3]
```diff
@@ -62,6 +65,8 @@ group :test do
   gem 'selenium-webdriver'
   # Easy installation and use of chromedriver to run system tests with Chrome
   gem 'chromedriver-helper'
+  gem 'database_cleaner'
```
Comment (Gmarn) [inline Gemfile]: Lock gem ;)

**Review by Gmarn — COMMENTED** [review]
Now we have travis you do not need to add screenshot from test ;)

File: Gemfile (hunk 2 above)
Comment (storresy) [reply Gemfile:48]: I also thought that gem should be in test group but rspec-rails documentation suggests we should put it in both development and test group 🤔 
https://github.com/rspec/rspec-rails#installation 

So, should I move that to test group anyway?

File: Gemfile (hunk 3 above)
Comment (storresy) [reply Gemfile]: fixed

File: .travis.yml (hunk 1 above)
Comment (storresy) [reply .travis.yml]: Just added that option. Much easier to debug this way. Thanks 😃

**Review by Gmarn — APPROVED** [review]

**Pull request merged**